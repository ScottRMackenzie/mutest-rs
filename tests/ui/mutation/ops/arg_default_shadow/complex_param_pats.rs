//@ print-mutants
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: arg_default_shadow

#![feature(decl_macro)]

#![allow(unused_variables)]

struct S<'a, T, D: Default + Copy> {
    v: Result<&'a str, usize>,
    t: T,
    d: D,
}

fn f<T, D: Default + Copy>((n, test): (usize, bool), s @ &S { v, ref t, d }: &S<'_, T, D>) {
    let _: usize = n;
    let _: bool = test;
    let _: Result<&str, usize> = v;
    let _: &T = t;
    let _: D = d;
    let _: &S<'_, T, D> = s;
}

#[test]
fn test() {
    f((1, true), &S { v: Ok("foo"), t: -1_i32, d: -1_i32 });
}
