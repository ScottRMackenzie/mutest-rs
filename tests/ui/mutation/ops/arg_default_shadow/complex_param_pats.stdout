1 mutation
  - [arg_default_shadow] ignore `n` argument by shadowing it with `Default::default()` in f at tests/ui/mutation/ops/arg_default_shadow/complex_param_pats.rs:13:28: 13:52
    <-(0)- test

1 mutation
  - [arg_default_shadow] ignore `test` argument by shadowing it with `Default::default()` in f at tests/ui/mutation/ops/arg_default_shadow/complex_param_pats.rs:13:28: 13:52
    <-(0)- test

1 mutation
  - [arg_default_shadow] ignore `d` argument by shadowing it with `Default::default()` in f at tests/ui/mutation/ops/arg_default_shadow/complex_param_pats.rs:13:54: 13:90
    <-(0)- test

3 mutants; 3 mutations; 3 safe; 0 unsafe (0 tainted); 0 batched; 3 unbatched
